src/helm/proxy/clients/palmyra_client.py

```diff
@@ -0,0 +1,125 @@
+import json
+import requests
+from typing import Any, Dict, List
+
+from helm.common.cache import Cache, CacheConfig
+from helm.common.request import Request, RequestResult, Sequence, Token
+from helm.common.tokenization_request import (
+    DecodeRequest,
+    DecodeRequestResult,
+    TokenizationRequest,
+    TokenizationRequestResult,
+)
+from .client import Client, wrap_request_time, truncate_sequence
+
+
+class PalmyraClient(Client):
+    def __init__(self, api_key: str, cache_config: CacheConfig, tokenizer_client: Client):
+        self.api_key: str = api_key
+        self.cache = Cache(cache_config)
+        self.tokenizer_client = tokenizer_client
+
+    def _send_request(self, model_name: str, raw_request: Dict[str, Any]) -> Dict[str, Any]:
+        response = requests.request(
+            method="POST",
+            url=f"https://enterprise-api.writer.com/llm/organization/3002/model/{model_name}/completions",
+            headers={
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+                "Authorization": f"Bearer {self.api_key}",
+            },
+            data=json.dumps(raw_request),
+        )
+        result = json.loads(response.text)
+        if "error" in result:
+            raise ValueError(f"Request failed with error: {result['error']}")
+        return result
+
+    def make_request(self, request: Request) -> RequestResult:
+        """Make a request"""
+        raw_request = {
+            "prompt": request.prompt,
+            "maxTokens": request.max_tokens,
+            "temperature": request.temperature,
+            "topP": request.top_p,
+            "bestOf": request.top_k_per_token,
+            "stop": request.stop_sequences,
+            "random_seed": request.random,
+        }
+
+        completions: List[Sequence] = []
+        model_name: str = request.model_engine
+
+        # `num_completions` is not supported, so instead make `num_completions` separate requests.
+        for completion_index in range(request.num_completions):
+            try:
+
+                def do_it():
+                    # HACKY: Use the random seed to get different results for each completion.
+                    raw_request["random_seed"] = (
+                        completion_index if request.random is None else request.random + completion_index
```

Comment:
Let's have the suffix be something like ":completion_index=0". This reduces the chance of users accidentally using our internal random seed (e.g. if users are setting random seed to "0" through "100")